mixed/flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/ArcticCatalog.java

```diff
@@ -343,8 +338,188 @@ private static void validateFlinkTable(CatalogBaseTable table) {
 
   @Override
   public void alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean ignoreIfNotExists)
-      throws CatalogException {
-    throw new UnsupportedOperationException();
+      throws CatalogException, TableNotExistException {
+    validateFlinkTable(newTable);
+
+    TableIdentifier tableIdentifier = getTableIdentifier(tablePath);
+    ArcticTable arcticTable;
+    try {
+      arcticTable = internalCatalog.loadTable(tableIdentifier);
+    } catch (org.apache.iceberg.exceptions.NoSuchTableException e) {
+      if (!ignoreIfNotExists) {
+        throw new TableNotExistException(internalCatalog.name(), tablePath, e);
+      } else {
+        return;
+      }
+    }
+
+    // Currently, Flink SQL only support altering table properties.
+    validateTableSchemaAndPartition(
+        toCatalogTable(arcticTable, tableIdentifier), (CatalogTable) newTable);
+
+    if (arcticTable.isUnkeyedTable()) {
```

Comment:
The code for altering properties seems very similar for both primary key tables and tables without primary keys. The only difference is the handling of some special properties. Maybe we can merge their implementations to reduce code redundancy.